Tdr 96 antivirus build

MancunianSam opened the pull request.

MancunianSam committed:
Add Jenkinsfiles

MancunianSam committed:
Add script block

MancunianSam committed:
Add a steps block

MancunianSam committed:
More messing with step blocks

MancunianSam committed:
Add stage name

MancunianSam committed:
Stop putting the script in a function

MancunianSam committed:
Remove method header from file

MancunianSam committed:
Small changes

MancunianSam committed:
More small changes

MancunianSam committed:
Don't specify the branch directly

MancunianSam committed:
Change parameter description

suzannehamilton commented on Dockerfile:
```diff
@@ -0,0 +1,2 @@
+FROM alpine
```
What does this Dockerfile do?

MancunianSam replied:
:smile: Absolutely nothing, I was testing something, I've deleted it.

MancunianSam committed:
Add Jenkinsfile for bundle build

I've added in a separate Jenkinsfile to create the lambda bundle, upload
it to S3 and deploy the lambda.
This needs to be called twice, once in the test multi branch job for the
master branch and once in the build job which updates the virus
definitions so it makes sense to have it in a separate job.

MancunianSam committed:
Update the build job to use the bundle job.

suzannehamilton reviewed (commented):
Let me know when you want a proper review - I'm guessing this PR will change based on this morning's conversation. Just one comment that I wrote before we had the chat.

MancunianSam committed:
Correct the name for jenkinslib

MancunianSam committed:
Add version tag parameter

MancunianSam committed:
Remove unused parameter.

MancunianSam committed:
Hardcode master instead of BRANCH_NAME

BRANCH_NAME doesn't exist on a standard pipeline job and we'll only ever
be deploying to master anyway so this is fine.

MancunianSam committed:
Add a steps block

MancunianSam committed:
Remove test Dockerfile

MancunianSam commented:
I've changed this now so there's a separate bundle job so this is ready for review now.

MancunianSam committed:
Add deploymentPackageName parameter

suzannehamilton commented on Jenkinsfile-bundle:
```diff
@@ -0,0 +1,64 @@
+        script {
+          sh "aws ecr get-login --region eu-west-2 --no-include-email | bash"
+          sh "docker stop dependencies | true"
+          sh "docker stop rules | true"
```
This might be a silly question, but why do we need to run `docker stop` during a build? Why would there be any Docker containers running at this point?

MancunianSam replied:
It does need to be run because this is running on the master Jenkins container, the container persists between builds so we do need to run `docker stop` to delete them. It doesn't really belong here though so I've moved it to a post block so it's cleaned up after each run.

suzannehamilton replied:
Cheers, I think that's easier to understand. 👍

suzannehamilton commented on Jenkinsfile-bundle:
```diff
@@ -0,0 +1,64 @@
+          sh "aws ecr get-login --region eu-west-2 --no-include-email | bash"
+          sh "docker stop dependencies | true"
+          sh "docker stop rules | true"
+          sh "rm -rf lambda"
```
Is this to clean up the output of a previous job? I'm wondering if it would be safer to set the Jenkins job to do a clean checkout each time?

MancunianSam replied:
Yep, there's a setting to delete all untracked files when the job runs, I've set this in the job settings and deleted the line to rm -rf lambda.

suzannehamilton reviewed (commented):
Just a couple of questions. Thanks for splitting up the Jenkins jobs - I think it's easier to follow the workflow now. 👍

MancunianSam committed:
Stop the docker containers in the post script

MancunianSam committed:
Add the lambda cleanup script in the post block

MancunianSam committed:
Remove the line to delete the folder
